client/src/featureform/client.py

```diff
@@ -188,6 +192,135 @@ def close(self):
         """
         self.impl.close()
 
+    def add_trigger(self, trigger, resource):
+        """
+        Add a trigger to a resource
+
+        **Example:**
+        ```py title="definitions.py"
+        client.add_trigger("trigger_name", ("resource_name", "resource_variant, resource_type"))
+        ```
+
+        Args:
+            trigger(Union[str, TriggerResource]): The name of the trigger
+            resource(Union[tuple, FeatureColumnResource, TrainingSetVariant]): The name, variant and type of the resource
+        """
+        req = metadata_pb2.TriggerRequest()
+
+        trigger_req = metadata_pb2.Trigger()
+        if isinstance(trigger, str):
+            trigger_req.name = trigger
+        elif isinstance(trigger, TriggerResource):
+            trigger_req.name = trigger.name
+        else:
+            raise ValueError("Invalid trigger type")
+        req.trigger.CopyFrom(trigger_req)
+
+        resource_req = metadata_pb2.ResourceID()
+        if isinstance(resource, tuple):
+            resource_req.resource.name = resource[0]
+            resource_req.resource.variant = resource[1]
+            # TODO: Need to convert this to a real resource type
+            resource_req.resource_type = resource[2]
+        elif isinstance(resource, Union[FeatureColumnResource, TrainingSetVariant]):
+            resource_req.resource.name = resource.name
+            resource_req.resource.variant = resource.variant
+            resource_req.resource_type = resource.resource_type
+        else:
+            raise ValueError("Invalid resource type")
+        req.resource.CopyFrom(resource_req)
+
+        self._stub.AddTrigger(req)
+    
+    def remove_trigger(self, trigger, resource):
+        """
+        Remove a trigger from a resource
+
+        **Example:**
+        ```py title="definitions.py"
+        client.remove_trigger("trigger_name", ("resource_name", "resource_variant, resource_type"))
+        ```
+
+        Args:
+            trigger(Union[str, TriggerResource]): The name of the trigger
+            resource(Union[tuple, FeatureColumnResource, TrainingSetVariant]): The name, variant and type of the resource
+        """
+        req = metadata_pb2.TriggerRequest()
+
+        trigger_req = metadata_pb2.Trigger()
+        if isinstance(trigger, str):
+            trigger_req.name = trigger
+        elif isinstance(trigger, TriggerResource):
+            trigger_req.name = trigger.name
+        else:
+            raise ValueError("Invalid trigger type")
+        req.trigger.CopyFrom(trigger_req)
+
+        resource_req = metadata_pb2.ResourceID()
+        if isinstance(resource, tuple):
+            resource_req.resource.name = resource[0]
+            resource_req.resource.variant = resource[1]
+            # TODO: Need to convert this to a real resource type
+            resource_req.resource_type = resource[2]
```

Comment:
Same as above, should check the tuple length so we dont go out of bounds with this. Ideally should just make a function that can be used in all these functions instead of repeating code